src/System.Runtime/tests/System/Reflection/AssemblyTests.cs

```diff
@@ -324,6 +328,11 @@ public void Test_LoadFrom()
             Assert.Throws<ArgumentNullException>("assemblyFile", () => Assembly.LoadFrom(null));
             var assem1 = Assembly.LoadFrom(destTestAssemblyPath);
             Assert.Equal(assem, assem1);
+
+            assem = Assembly.LoadFrom(typeof(AssemblyTests).Assembly.Location);
```

Comment:
You should add a comment here explaining why loading an assembly from two different paths resolves to the same assembly reference.

LGTM otherwise.